### PR TITLE
feat: Create USSF Documentation page

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-543

## Proposed changes
- Adds new page `/ussf-documentation`
- Moves Essential Reading docs from `/about-us` to `/ussf-documentation`
- Updates `<EPubsCard />` component to make the `query` prop optional and adds an optional `title` prop
- Also updated e2e tests [here](https://github.com/USSF-ORBIT/ussf-portal/pull/164)

---

## Reviewer notes
- Verify that 'USSF documentation' is now an option in the side navigation
- Navigate to `/ussf-documentation` and verify that it matches the below screenshot
- Toggle the Accordion component and verify that the 'Essential Reading' docs are present
- Click the link in the ePubs box and verify that it takes you to the ePubs search page
- Toggle light/dark mode and verify that the styles match the [design in Figma](https://www.figma.com/file/iPOdDkRxCaeoDoQ4JXjxlk/Space-Force-Service-Portal?node-id=10866%3A63503&t=3q1TgiCSUHoyGglu-0)

## Setup
As always, run both the portal client and the CMS on your local machine

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [x] Checked in the design translated visually
- [x] Checked behavior
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links
- [x] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [x] Checked responsiveness in mobile, tablet, and desktop
  - [x] Checked keyboard navigability
  - [x] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [x] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

## Screenshots

docs-page

### CODE REVIEW

**Comment by jbecker01** [comment]
I noticed on Light Mode, the toggle button color is #243593 instead of #82CC92

Screen Shot 2022-11-14 at 3 27 46 PM

Screen Shot 2022-11-14 at 3 27 55 PM

**Review by malworks — CHANGES_REQUESTED** [review]
Hi! Looks great in light and dark mode. A couple things:

- The accordion headers ("Essential Reading") are using h3's, when they should be using the label tag. 
image

- Question: The design breaks at more narrow tablet/mobile widths but I'm not sure we're "officially" supporting mobile or not. Are we?
- Figma has the document type icon listed in front of each document. Is this in scope? (See photo below) 

image

File: src/__tests__/pages/ussf-documentation.test.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  describe('when logged in', () => {
+    let html: RenderResult
+
+    beforeEach(() => {
+      html = renderWithAuth(
+        <MockedProvider>
+          <USSFDocumentation />
+        </MockedProvider>
+      )
+    })
```
Comment (gidjin) [inline src/__tests__/pages/ussf-documentation.test.tsx:64]: Thought, is there a reason to continue this pattern of sharing `html`. IIRC the recommendation is to not do that now with react-testing-library. I think this was the article: https://kentcdodds.com/blog/avoid-nesting-when-youre-testing

Suggestion: refactor this to call render in each it block or just combine them all into one test. I can't find the article that recommended that but I read one from Kent Dodds about that.

**Review by gidjin — APPROVED** [review]
Minor note about the test structure in the new test file, but I don't see a need to block the PR on that. Approving from an eng stand point, please follow up with Mallory on the notes she left.

Question: Should the `Essential Reading` section be expanded by default since it's the only one? Might be a good question for @malworks or @shkeating

**Comment by jcbcapps** [comment]
> Hi! Looks great in light and dark mode. A couple things:
> 
> * The accordion headers ("Essential Reading") are using h3's, when they should be using the label tag.
> 
> * Question: The design breaks at more narrow tablet/mobile widths but I'm not sure we're "officially" supporting mobile or not. Are we?
> * Figma has the document type icon listed in front of each document. Is this in scope? (See photo below)
> 
@malworks 
1) Good catch on the h3's, I'll make that change.
2) We don't currently support mobile and, as far as I know, we have no concrete plans to do so.
3) I skipped this for now because we're still working on the file upload functionality. I plan to write a chore in Shortcut to loop back and move these files to the CMS after we get file upload done.

**Comment by jcbcapps** [comment]
@malworks I spoke too soon on changing the h3 in the Accordion. Guess I should have mentioned that the Accordion component is a ReactUSWDS component, and it uses a `headingLevel` prop for each item that is required. So I can't make that change, but it doesn't seem crucial to me. Any thoughts? 

Also, @gidjin had a great question above asking if the 'Essential Reading' section should be open by default since it is the only Accordion item. I went ahead and made that change so that you could see what it would look like, but let me know if I need to change it back!

**Comment by malworks** [comment]
> @malworks I spoke too soon on changing the h3 in the Accordion. Guess I should have mentioned that the Accordion component is a ReactUSWDS component, and it uses a `headingLevel` prop for each item that is required. So I can't make that change, but it doesn't seem crucial to me. Any thoughts?
> 
> Also, @gidjin had a great question above asking if the 'Essential Reading' section should be open by default since it is the only Accordion item. I went ahead and made that change so that you could see what it would look like, but let me know if I need to change it back!

No worries, it's fine as is. And nice addition to keep the 'Essential Reading' open by default!

**Review by malworks — APPROVED** [review]

**Pull request merged**